File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    dict = {}
    for i in participant:
        if dict.get(i) == None:
            dict[i] = 1
        else:
            dict[i] += 1
        
    for i in completion:
        dict[i] -= 1
    
    
    for i in dict:
        if dict[i] > 0:
            return i
                

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
  
    map = new Map()
    for (let name of completion){
        map.set(name,(map.get(name)?? 0) + 1)
    }
    
    for(let name of participant){
        if(map.get(name) > 0){
            map.set(name,map.get(name) -1)
            continue 
        }
        return name 
    }


}